flink-tests/src/test/java/org/apache/flink/test/completeness/TypeSerializerTestCoverageTest.java

```diff
@@ -0,0 +1,180 @@
+				ValueSerializer.class.getName(),
+				RowSerializer.class.getName(),
+				StreamElementSerializer.class.getName(),
+				WritableSerializer.class.getName(),
+				KryoSerializer.class.getName(),
+				UnloadableDummyTypeSerializer.class.getName(),
+				TupleSerializer.class.getName(),
+				EnumSerializer.class.getName(),
+				CoGroupedStreams.UnionSerializer.class.getName(),
+				TtlStateFactory.TtlSerializer.class.getName(),
+				TimeWindow.Serializer.class.getName(),
+				InternalTimersSnapshotReaderWriters.LegacyTimerSerializer.class.getName(),
+				TwoPhaseCommitSinkFunction.StateSerializer.class.getName(),
+				IntervalJoinOperator.BufferEntrySerializer.class.getName(),
+				GlobalWindow.Serializer.class.getName(),
+				org.apache.flink.queryablestate.client.VoidNamespaceSerializer.class.getName(),
+				org.apache.flink.runtime.state.VoidNamespaceSerializer.class.getName()
+		);
+
+		//  type serializer whitelist for TypeSerializerUpgradeTestBase test coverage
+		final List<String> typeSerializerUpgradeTestBaseWhitelist = Arrays.asList(
+				KeyAndValueSerializer.class.getName(),
+				InstantSerializer.class.getName(),
+				SingleThreadAccessCheckingTypeSerializer.class.getName(),
+				SimpleVersionedSerializerTypeSerializerProxy.class.getName(),
+				Tuple0Serializer.class.getName(),
+				CopyableValueSerializer.class.getName(),
+				VoidSerializer.class.getName(),
+				StringArraySerializer.class.getName(),
+				LocalDateTimeSerializer.class.getName(),
+				BooleanPrimitiveArraySerializer.class.getName(),
+				BytePrimitiveArraySerializer.class.getName(),
+				CharPrimitiveArraySerializer.class.getName(),
+				DoublePrimitiveArraySerializer.class.getName(),
+				FloatPrimitiveArraySerializer.class.getName(),
+				IntPrimitiveArraySerializer.class.getName(),
+				LongPrimitiveArraySerializer.class.getName(),
+				ShortPrimitiveArraySerializer.class.getName(),
+				LocalDateSerializer.class.getName(),
+				LocalTimeSerializer.class.getName(),
+				UnloadableDummyTypeSerializer.class.getName(),
+				TimeWindow.Serializer.class.getName(),
+				CoGroupedStreams.UnionSerializer.class.getName(),
+				InternalTimersSnapshotReaderWriters.LegacyTimerSerializer.class.getName(),
+				TwoPhaseCommitSinkFunction.StateSerializer.class.getName(),
+				GlobalWindow.Serializer.class.getName()
+		);
+
+		// check if a test exists for each type serializer
+		for (Class<? extends TypeSerializer> typeSerializer : typeSerializers) {
+			// we skip abstract classes, test classes, inner classes and scala classes to skip type serializer defined in test classes
+			if (Modifier.isAbstract(typeSerializer.getModifiers()) ||
+					Modifier.isPrivate(typeSerializer.getModifiers()) ||
+					typeSerializer.getName().contains("test") ||
+					typeSerializer.getName().contains("Test") ||
+					typeSerializer.getName().contains("ITCase") ||
+					typeSerializer.getName().contains("$$anon") ||
+					typeSerializer.getName().contains("scala")) {
```

Comment:
These exclusions are a bit too broad now. Like this, newly added Scala `TypeSerializers` will not trigger the test.